CMakeLists.txt

```diff
@@ -867,209 +906,60 @@ SET(ALC_OBJS  ${ALC_OBJS}
 )
 
 # Check ALSA backend
-OPTION(ALSOFT_REQUIRE_ALSA "Require ALSA backend" OFF)
-FIND_PACKAGE(ALSA)
-IF(ALSA_FOUND)
-    OPTION(ALSOFT_BACKEND_ALSA "Enable ALSA backend" ON)
-    IF(ALSOFT_BACKEND_ALSA)
-        SET(HAVE_ALSA 1)
-        SET(BACKENDS  "${BACKENDS} ALSA${IS_LINKED},")
-        SET(ALC_OBJS  ${ALC_OBJS} Alc/backends/alsa.c)
-        ADD_BACKEND_LIBS(${ALSA_LIBRARIES})
-        IF(CMAKE_VERSION VERSION_LESS "2.8.8")
-            INCLUDE_DIRECTORIES(${ALSA_INCLUDE_DIRS})
-        ENDIF()
-    ENDIF()
-ENDIF()
-IF(ALSOFT_REQUIRE_ALSA AND NOT HAVE_ALSA)
-    MESSAGE(FATAL_ERROR "Failed to enabled required ALSA backend")
+ADD_BACKEND(ALSA ALSA Alc/backends/alsa.c)
+IF(ALSOFT_BACKEND_ALSA)
```

Comment:
I'm not sure these should check the ```ALSOFT_BACKEND_*``` options, since they may be cached values and will stay even if the backend is no longer available. I think it makes more sense to check the ```HAVE_*``` variables.